x-pack/plugins/transform/public/app/sections/transform_management/components/action_clone/use_clone_action.tsx

```diff
@@ -11,32 +11,32 @@ import { i18n } from '@kbn/i18n';
 
 import type { TransformListAction, TransformListRow } from '../../../../common';
 import { SECTION_SLUG } from '../../../../common/constants';
-import { useTransformCapabilities, useSearchItems } from '../../../../hooks';
-import { useAppDependencies, useToastNotifications } from '../../../../app_dependencies';
+import { useGetDataViewIdsWithTitle, useTransformCapabilities } from '../../../../hooks';
+import { useToastNotifications } from '../../../../app_dependencies';
 
 import { cloneActionNameText, CloneActionName } from './clone_action_name';
 
 export type CloneAction = ReturnType<typeof useCloneAction>;
 export const useCloneAction = (forceDisable: boolean, transformNodes: number) => {
   const history = useHistory();
-  const appDeps = useAppDependencies();
-  const dataViewsContract = appDeps.data.dataViews;
   const toastNotifications = useToastNotifications();
 
-  const { getDataViewIdByTitle, loadDataViews } = useSearchItems(undefined);
-
+  const { data: dataViewListItems } = useGetDataViewIdsWithTitle();
   const { canCreateTransform } = useTransformCapabilities();
 
   const clickHandler = useCallback(
     async (item: TransformListRow) => {
       try {
-        await loadDataViews(dataViewsContract);
+        if (!dataViewListItems) {
+          return;
+        }
+
         const dataViewTitle = Array.isArray(item.config.source.index)
           ? item.config.source.index.join(',')
           : item.config.source.index;
-        const dataViewId = getDataViewIdByTitle(dataViewTitle);
+        const dataViewListItem = dataViewListItems.find((d) => d.title === dataViewTitle);
```

Comment:
Each of the uses of `dataViewListItems` runs a `find`, these won't be too expensive, but it might be slightly more efficient if `useGetDataViewIdsWithTitle` returned a map of data views, keyed on `title`

Reply:
👍  Updated the hook to return a map, also renamed everything to be in line with the new returned structure in [9f23cb8](https://github.com/elastic/kibana/pull/181892/commits/9f23cb895147f7cac6d1ab06df8f7a0e39cb1f78).